server/snapshot.go

```diff
@@ -234,6 +246,27 @@ func (r *raftFSM) Restore(snapshot io.ReadCloser) (retErr error) {
 	r.Lock()
 	defer r.Unlock()
 
+	shouldSnapshot := false
+	defer func() {
+		if retErr == nil && shouldSnapshot {
```

Comment:
Doesn't this snapshot retErr here when defining the func? Meaning if you set it later it won't matter? May be remembering incorrectly.

Reply:
No, what is deferred is the invocation of the function. So function will execute and evaluate whatever is inside. I have tested it works ok.

Reply:
ok